infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationClientConfigListener.java

```diff
@@ -0,0 +1,218 @@
+     */
+    private void handleClientMetadataTableEntries(List<CorfuStreamEntry> clientMetadataTableEntries) {
+        Set<LogReplicationSession> impactedSessions = new HashSet<>();
+        log.info("clientMetadataTableEntries: {}", clientMetadataTableEntries);
+        if (clientMetadataTableEntries == null) {
+            log.warn("No client metadata table entries found!");
+            return;
+        }
+
+        for (CorfuStreamEntry entry : clientMetadataTableEntries) {
+            ClientDestinationInfoKey clientInfo = (ClientDestinationInfoKey) entry.getKey();
+            LogReplication.ReplicationSubscriber subscriber = LogReplication.ReplicationSubscriber.newBuilder()
+                    .setClientName(clientInfo.getClientName()).setModel(clientInfo.getModel()).build();
+            if (entry.getOperation().equals(CorfuStreamEntry.OperationType.UPDATE)) {
+                DestinationInfoVal sinksInfo = (DestinationInfoVal) entry.getPayload();
+                impactedSessions = configManager.onGroupDestinationsChange(subscriber, clientInfo.getGroupName(),
+                        sinksInfo.getDestinationIdsList());
+            } else if (entry.getOperation().equals(CorfuStreamEntry.OperationType.DELETE)) {
+                impactedSessions = configManager.onGroupDestinationsChange(subscriber, clientInfo.getGroupName(),
+                        new ArrayList<>());
+            }
+
+            if (impactedSessions != null) {
+                log.info("Sessions that a forced snapshot sync will be triggered: {}", impactedSessions);
+                impactedSessions.forEach(session -> {
+                    SnapshotSyncUtils.enforceSnapshotSync(session, corfuStore);
+                });
+            }
+        }
+    }
+
+    /**
+     * Unsubscribe this stream listener to stop
+     */
+    public void stop() {
+        corfuStore.unsubscribeListener(this);
+        started.set(false);
+    }
+
+    /**
+     * Check if the listener is started.
+     * @return true if started, false otherwise.
+     */
+    public boolean listenerStarted() {
+        return started.get();
+    }
+
+    /**
+     * Perform full sync (snapshot) on a table or tables of interest (the ones subscribed to).
+     *
+     * @return timestamp from which to subscribe for deltas.
+     * Note: this timestamp should come from txn.commit() of the full sync.
+     */
+    @Override
+    protected CorfuStoreMetadata.Timestamp performFullSync() {
+        // TODO (V2 / Chris): In next PR this listener will be only for client register/unregister, remember to avoid
+        //  clearing all the in-memory fields in onClientListenerResume method in next PR.
+        CorfuStoreMetadata.Timestamp timestamp = configManager.onClientListenerResume();
+        configManager.generateConfig(sessionManager.getSessions());
+        sessionManager.getSessions().forEach(session -> {
+            SnapshotSyncUtils.enforceSnapshotSync(session, corfuStore);
```

Comment:
Maybe I missed something but won't this generate a forced snapshot sync even upon restart - let's say LR server just bounced - we will call this method and it will figure out the sessions and call force snapshot sync right?

Reply:
I think this method `performFullSync()` will be executed only in error path. For LR restart it should go with the normal start listener path?